### ISSUE TITLE
AheadBy and BehindBy on Prompt

### ISSUE THREAD

**Issue by terryaney** [dahlbyk/posh-git#203]
They seem to have been removed?  An older version of prompt had these lines (unless I always put them in manually :S can't remember) right before `if($s.EnableFileStatus -and $status.HasIndex) {`

```
if ($status.BehindBy -gt 0) {
    # We are behind remote
    Write-Prompt "(-" -BackgroundColor $branchBackgroundColor -ForegroundColor $branchForegroundColor
    Write-Prompt $status.BehindBy -BackgroundColor $branchBackgroundColor -ForegroundColor $branchForegroundColor
    Write-Prompt ")" -BackgroundColor $branchBackgroundColor -ForegroundColor $branchForegroundColor
}
if ($status.AheadBy -gt 0) {
    # We are ahead of remote
    Write-Prompt "(+" -BackgroundColor $branchBackgroundColor -ForegroundColor $branchForegroundColor
    Write-Prompt $status.AheadBy -BackgroundColor $branchBackgroundColor -ForegroundColor $branchForegroundColor
    Write-Prompt ")" -BackgroundColor $branchBackgroundColor -ForegroundColor $branchForegroundColor
}
```

Did I add those manually or were they in fact removed?

**Comment by paulmarsy** [comment]
I can't remember seeing that functionality in the past however I have proposed PR #202 which would change the display of the branch's status if you have any feedback on how the information is displayed.

The current PR doesn't show how far ahead/behind the branch is.. my feeling is that it would likely add too much noise. How useful do you find knowing the amount that the branch is ahead/behind?

**Comment by dahlbyk** [comment]
I don't recognize those changes @terryaney - they may, in fact, be yours. I've finally merged #202, and am also curious how useful you have found the ahead/behind counts to be.

**Comment by terryaney** [comment]
Hmm...don't recall editing the files, but I guess I could have.  I find them use full.  Have 100s of private (client) repos that I'm cycling through.  When in command prompt and I switch to a directory, it is nice/easy cue as to whether I need to push changes if it has a +N in the prompt.  Additionally, I have a C# script that walks all folders doing an 'audit' of my GIt repos to see if any need publishing and they use those values as well to determine if I need to push (which is useful).

**Comment by dahlbyk** [comment]
> When in command prompt and I switch to a directory, it is nice/easy cue as to whether I need to push changes if it has a +N in the prompt.

With #202 merged you'll now see ↑ or ↕ anywhere you would have seen +N, you're just missing the actual count. For my part, that count is only interesting insofar as it is nonzero (communicated by arrows).


> Additionally, I have a C# script that walks all folders doing an 'audit' of my GIt repos to see if any need publishing and they use those values as well to determine if I need to push (which is useful).

Are you using LibGit2Sharp for this or shelling out to Git? I wouldn't expect a change in posh-git to impact your script.

**Comment by terryaney** [comment]
Reviewing my LINQ script...I do indeed just shell out git and determine ahead/behind by parsing the git status text (is ahead/is behind)...so I should be good.  Will give new posh-git a try, thanks.

**Comment by drawfour** [comment]
I've created pull request #256, which makes the prompt configurable for how it shows ahead/behind in the prompt.

> and am also curious how useful you have found the ahead/behind counts to be.

I find it really useful.  If I'm behind by a lot, it means I haven't pulled in a while, and I probably should do it soon.  If the count is pretty low, I'm fairly recent and don't have to worry.  If I'm ahead by a lot, then I probably have some commits I need to squash or should go check my outstanding code reviews for feedback/signoff to get them in.

It's really great when switching to a branch to see exactly how far behind or ahead I am.

**Comment by rkeithhill** [comment]
This issue has been addressed by PR #256 which was merged recently. It will be enabled by default in the upcoming 0.7.0 release.
